docs/DevGuide/ImportingData.md

```diff
@@ -0,0 +1,60 @@
+\cond NEVER
+Distributed under the MIT License.
+See LICENSE.txt for details.
+\endcond
+# Importing data {#dev_guide_importing}
+
+The `importers` namespace holds functionality for importing data into SpECTRE.
+We currently support loading volume data files in the same format that is
+written by the `observers`.
+
+## Importing volume data
+
+The `importers::VolumeDataReader` parallel component is responsible for loading
+volume data and distributing it to elements of one or multiple array parallel
+components. As a first step, make sure you have added the
+`importers::VolumeDataReader` to your `Metavariables::component_list`. Also make
+sure you have a `Metavariables::Phase` in which you will perform registration
+with the importer, and another for loading the data. Here's an example for such
+`Metavariables`:
+
+\snippet Test_VolumeDataReaderAlgorithm.hpp metavars
+
+To register elements of your array parallel component for receiving volume data,
+invoke the `importers::Actions::RegisterWithVolumeDataReader` action in the
+`phase_dependent_action_list`. Here's an example from the `ElementArray`
+component mentioned in the previous code snippet:
+
+\snippet Test_VolumeDataReaderAlgorithm.hpp register_action
+
+Now you're all set to load a data file. To do so, invoke the
+`importers::ThreadedActions::ReadVolumeData` action on the
+`importers::VolumeDataReader`. The action takes template parameters that
+determine the data file and fields to load. It also takes an action that will be
+invoked to receive the data on each element of the array component. Here's an
+example from the `ElementArray` component that assembles such an action:
+
+\snippet Test_VolumeDataReaderAlgorithm.hpp read_data_action
+
+- The `VolumeDataOptions` in this example is an option group that supplies
+  information such as the file name. You provide an option group that represents
+  the data you want to import. For example, we have the
+  `evolution::OptionTags::NumericInitialData` that represents numeric initial
+  data for an evolution. In our example, we created a new class like this:
+
+  \snippet Test_VolumeDataReaderAlgorithm.hpp option_group
+
+  This results in a section in the input file that may look like this:
+
+  \snippet Test_VolumeDataReaderAlgorithm2D.yaml importer_options
+
+- The `::Actions::SetData` callback action in the example will be invoked to
+  receive the data on each element. You can use this particular action to move
+  the received data directly into the DataBox, but it's generally better to
+  write a new action that does a few consistency checks on the data before
+  moving it into the DataBox.
+
+Once you have assembled the `importers::ThreadedActions::ReadVolumeData` action
```

Comment:
I don't understand what you mean by "assemble". What's not clear to me from this documentation is how to actually get the data loaded onto the ElementArray. The registration part is clear, but then for the phase `ImportData` I don't know what to do.

Reply:
This is written earlier in the docs:

> Here's an example from the ElementArray component that assembles such an action:
>
> ```cpp
> using import_fields = tmpl::list<ScalarFieldTag, VectorFieldTag<Dim>>;
> using read_element_data_action = importers::ThreadedActions::ReadVolumeData<
>      VolumeDataOptions<TheGrid>, import_fields,
>      ::Actions::SetData<import_fields>, ElementArray>;
> ```

Then I explain some of the template parameters, and then:

> Once you have assembled the importers::ThreadedActions::ReadVolumeData action you can invoke it like this:
>
> ```c++
>  static void execute_next_phase(
>       const typename Metavariables::Phase next_phase,
>       Parallel::CProxy_ConstGlobalCache<Metavariables>& global_cache) noexcept {
>     auto& local_cache = *(global_cache.ckLocalBranch());
>     Parallel::get_parallel_component<ElementArray>(local_cache)
>         .start_phase(next_phase);
>     if (next_phase == Metavariables::Phase::ImportData) {
>       Parallel::threaded_action<read_element_data_action>(
>           Parallel::get_parallel_component<
>               importers::VolumeDataReader<Metavariables>>(local_cache));
>     }
>   }
> ```

So with "assemble" I mean instantiate the `ReadVolumeData` action template with a bunch of carefully chosen template parameters. And I thought the snippet makes it quite clear how to invoke the action.

I added another paragraph which hopefully gives a better overview of what's going on.

Reply:
Oh, I see now what I don't understand. You should combine the `using import_fields`... snippet and the last snippet into one, then walk through what it does. The giant two bullets in the middle makes it difficult to associate the code. You could also leave the snippets separate and put the last one first, explaining how to invoke the volume read, then talking about what the action does. I think there is way too much info between the random type aliases that I thought were some special things the actions search for in the metavariables and the actual invokation could.

Reply:
I see the confusion. Reformulated as you suggested in the last commit. Is it clearer now?